core/src/main/java/nl/nn/adapterframework/doc/IbisDocPipe.java

```diff
@@ -15,328 +15,30 @@
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.extensions\\.ifsa\\.IfsaProviderListener");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.extensions\\.sap\\.jco2\\.SapSender");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.extensions\\.sap\\.jco2\\.SapListener");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.extensions\\.sap\\.jco2\\.SapLUWManager");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.extensions\\.sap\\.jco3\\.SapSender");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.extensions\\.sap\\.jco3\\.SapListener");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.extensions\\.sap\\.jco3\\.SapLUWManager");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.pipes\\.CommandSender");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.pipes\\.EchoSender");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.pipes\\.FixedResultSender");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.pipes\\.LogSender");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.pipes\\.MailSender");
-		excludeFilters.add(".*\\.IbisstoreSummaryQuerySender");
-		// Exclude classes that cannot be used directly in configurations
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.pipes\\.MessageSendingPipe");
-		
-		// Exclude classes that should only be used in internal configurations
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.doc\\.IbisDocPipe");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.webcontrol\\..*");
-		excludeFilters.add("nl\\.nn\\.adapterframework\\.pipes\\.CreateRestViewPipe");
-	}
-	private static Map<String, String> ignores = new HashMap<String, String>();
-	static {
-		// Adding a sender to a listener has been used in the past but is not (commonly) used anymore
-		ignores.put("Listener", "Sender");
-	}
-	private static List<String> excludeNameAttribute = new ArrayList<String>();
-	static {
-		excludeNameAttribute.add("putValidator");
-		excludeNameAttribute.add("putWrapper");
-	}
-	private static List<String> overwriteMaxOccursToOne= new ArrayList<String>();
-	static {
-		// When registerPipeLine in Adapter has been renamed to setPipeLine this workaround can be removed.
-		overwriteMaxOccursToOne.add("registerPipeLine");
-	}
-	private static List<String> overwriteMaxOccursToUnbounded = new ArrayList<String>();
-	static {
-		// The setSender in ParallelSenders adds senders to a list. When setSender  has been renamed to addSender this
-		// workaround can be removed.
-		overwriteMaxOccursToUnbounded.add("parallelSendersSender");
-	}
-	// Influence the order of elements in the XSD, this will override the alphabetic order.
-	// Instead of using the Maps below it might be a good idea to use annotations on the specified methods.
-	private static Map<String, Integer> sortWeightAdapter = new HashMap<String, Integer>();
-	static {
-		sortWeightAdapter.put("registerReceiver", 100);
-	}
-	private static Map<String, Integer> sortWeightReceiver = new HashMap<String, Integer>();
```

Comment:
Klopt het dat deze sortWeigths niet meer gebruikt worden?

Reply:
Ik heb de sortWeights ook verplaatst naar nl.nn.adapterframework.doc.InfoBuilderSource. Ik zal vertellen waarom ik deze class in het leven heb geroepen. Toen ik met file IbisDocPipe werkte, merkte ik dat de code boven method doPipe() en de code vanaf doPipe() aparte eenheden vormen. De code boven doPipe() had het karakter van utilities, terwijl het gebruik van die utilities vanaf doPipe() was. Het leek me overzichtelijk om dit expliciet te maken door een aparte klasse te maken. In de code die ik gemaakt heb is duidelijker welke stappen worden genomen om de DocInfo te vullen. Details over hoe dat gebeurt, zoals lange lijsten waardes en Spring details staan nu apart.